### PR TITLE
Mike/add filter for library

### PR DESCRIPTION
**Pull request by mikemtdev**
### This PR adds: 
- Grades
- Subjects
- Units
- Topics
- Categories

image

#### Showing no Items for filter
image

#### Showing items for filter
image

### CODE REVIEW

**Comment by mikemtdev** [comment]
> So eventually we are getting rid of courses, we will have the following filters
> 
>     * Grades
> 
>     * Subjects
> 
>     * Units
> 
>     * Topics
> 
> 
> The tags at the top will just be categories of the media content.
> 
> You can refer to the section here https://www.alemhulu.com
> 
> @brianmuks this means we should be able to query media contents Just by Grade, or by Subjects or by Units or Topics or Category in case there's need for BE support.

So this means there will be two places where subjects will appear?

File: src/app/api/media-content/index.ts [hunk 1]
```diff
@@ -51,7 +54,8 @@ export default async function fetchMediaContent_(request: any) {
     if (course_id) query.course_id = new BSON.ObjectId(course_id);
     if (unit_id) query.unit_id = new BSON.ObjectId(unit_id);
     if (topic_id) query.topic_id = new BSON.ObjectId(topic_id);
-
+    if (grade_id) query.grade_id = new BSON.ObjectId(grade_id);
+    console.debug('query', query);
```
[the remark this answers is not in the document]
Comment (mikemtdev) [reply src/app/api/media-content/index.ts:58]: Thanks will remove. 
I do think it would be best adding a rule. 
Just in case.

**Comment by brianmuks** [comment]
> So eventually we are getting rid of courses, we will have the following filters
> 
> * Grades
> * Subjects
> * Units
> * Topics
> 
> The tags at the top will just be categories of the media content.
> 
> You can refer to the section here https://www.alemhulu.com
> 
> @brianmuks this means we should be able to query media contents Just by Grade, or by Subjects or by Units or Topics or Category in case there's need for BE support.

I will look into this.

File: src/app/api/media-content/index.ts [hunk 2]
```diff
@@ -21,10 +21,13 @@ export default async function fetchMediaContent_(request: any) {
     course_id: z.string().optional(),
     unit_id: z.string().optional(),
     topic_id: z.string().optional(),
+    grade_id: z.string().optional(),
```
Comment (brianmuks) [inline src/app/api/media-content/index.ts:24]: @mikemtdev  I noticed you might be working on the backend for too.

While I'm happy to offer assistance if needed, I wanted to highlight the importance of maintaining clear documentation for future updates.

Unsolicited changes to the backend code can sometimes make it difficult to keep the frontend and backend in sync.

To ensure smooth collaboration and long-term maintainability, would you be open to discussing any planned backend changes before implementation? This way, we can keep the documentation up-to-date and avoid potential issues down the line.

Let me know your thoughts, and I'm happy to chat further!

**Review by brianmuks — CHANGES_REQUESTED** [review]

File: src/app/api/media-content/index.ts (hunk 2 above)
Comment (brianmuks) [reply src/app/api/media-content/index.ts:24]: @OlivierJM I am currently documenting end points in postman. @mikemtdev What do you need from the backend with regards to this `PR` ?

**Comment by mikemtdev** [comment]
> @mikemtdev let us know if you are blocked on this or if there is anything we can help with. 

No just need to remove backend files that I touched/added. 
Will make changes

**Comment by mikemtdev** [comment]
> @mikemtdev as part of this PR, can we a way to mark a filter that is currently selected(some active state like on dashboard), e.g: a user should know that they selected Grade 10, some params may help `?grade=10&subject=chemistry&topic=something` This way it is easier also to persist urls and even easier to come back to it.

I added something similar. This is via a hook called useSearchQuery. 
it will append to the URL string 
image

When you add a topic to the grade filter
image

**Comment by mikemtdev** [comment]
> > I added something similar. This is via a hook called useSearchQuery.
> 
> Yes this is good, do we get active items that were selected or this will be done in a separate issue, I would love to merge this sooner.

Yes, it shows. 
image

**Pull request merged**